### PR TITLE
Adds bespin-job-watcher service to job-details component

### PR DESCRIPTION
**Pull request by johnbradley**
This service updates the jobs in the store when a message comes in.
This is only used on the job-details screen but could be used elsewhere.

Also adds jobErrors to job model simplifying updates to this model.
Requires https://github.com/Duke-GCB/bespin-api/pull/31.

### CODE REVIEW

File: app/components/job-detail.js [hunk 1]
```diff
@@ -1,20 +1,51 @@
...
+    var watcher = this.get('watcher');
+    var token = this.getAuthToken();
+    var jobId = this.getJobId();
+    watcher.startWatching(token, jobId);
+  },
+  willDestroyElement() {
+    this._super(...arguments);
+    var watcher = this.get('watcher');
+    var token = this.getAuthToken();
+    var jobId = this.getJobId();
```
Comment (dleehr) [inline app/components/job-detail.js]: Any reason this isn't simply `var jobId = this.get('job.id');`?

Also, these `var`s can be `const`s, no?

File: app/routes/jobs/show.js [hunk 2]
```diff
@@ -3,24 +3,5 @@ import Ember from 'ember';
...
-      So it's more consistent for the job to be the single, primary model on this route, and load the errors after
-      the model is fulfilled.
-     */
-    this._super(controller, job);
-    // Fetch the job errors
-    this.get('store').query('job-error', {
-      job: job.get('id')
-    }).then((jobErrors) => {
-      controller.set('jobErrors', jobErrors);
-    });
```
Comment (dleehr) [inline app/routes/jobs/show.js]: Yay for removing my hacky workarounds that needed lots of comments.

File: config/environment.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ module.exports = function(environment) {
     // ENV.APP.LOG_VIEW_LOOKUPS = true;
     ENV.APP.API_HOST = 'http://127.0.0.1:8000';
     ENV.APP.API_NAMESPACE = 'api';
+    ENV.APP.JOB_WATCHER_HOST = 'localhost:8080';
```
Comment (dleehr) [inline config/environment.js]: Somewhat inconsistent that the `API_HOST` includes the protocol, but the `JOB_WATCHER_HOST` does not.

File: app/models/job.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ export default DS.Model.extend({
     return this.get('state') === 'F';
   }),
   outputDir: DS.belongsTo('job-output-dir'),
+  jobErrors: DS.hasMany('job-error'),
```
Comment (dleehr) [inline app/models/job.js]: I was going to suggest using simply `errors` as the field name, but now I see that's reserved for errors at the model level. Maybe add a comment here so we remember why this isn't called simply `errors`

**Comment by johnbradley** [comment]
@dleehr Any more thoughts on this one?

**Comment by dleehr** [comment]
Nope

**Pull request merged**